src/components/Contacts/ContactDetails/ContactDetailsTab/ContactDetailsTab.tsx

```diff
@@ -65,90 +98,187 @@ export const ContactDetailsTab: React.FC<ContactDetailTabProps> = ({
   });
 
   const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { push, query } = useRouter();
+  const [deleteContact, { loading: deleting }] = useDeleteContactMutation();
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
 
-  return (
-    <ContactDetailsTabContainer>
-      {
-        // Tag Section
-      }
-      <ContactDetailHeadingContainer>
-        {loading || !data ? (
-          <ContactDetailLoadingPlaceHolder variant="rect" />
-        ) : (
-          <ContactTags
-            accountListId={accountListId}
-            contactId={data.contact.id}
-            contactTags={data.contact.tagList}
-          />
-        )}
-      </ContactDetailHeadingContainer>
-      <Divider />
-      {
-        // People Section
-      }
-      <ContactDetailSectionContainer>
-        <ContactDetailHeadingContainer>
-          <ContactDetailHeadingText variant="h6">
-            {loading || !data ? t('Loading') : data.contact.name}
-          </ContactDetailHeadingText>
-          <ContactDetailHeadingIcon />
-        </ContactDetailHeadingContainer>
-        {loading || !data ? (
+  const { searchTerm } = query;
+
+  const handleDeleteContact = () => {
+    try {
+      deleteContact({
+        variables: {
+          accountListId,
+          contactId,
+        },
+        update: (cache, { data: deletedContactData }) => {
+          const deletedContactId = deletedContactData?.deleteContact?.id;
+          const query = {
+            query: ContactsDocument,
+            variables: {
+              accountListId,
+              searchTerm,
+            },
+          };
+
+          const dataFromCache = cache.readQuery<ContactsQuery>(query);
+
+          if (dataFromCache) {
+            const data = {
+              ...dataFromCache,
+              contacts: {
+                ...dataFromCache.contacts,
+                nodes: dataFromCache.contacts.nodes.filter(
+                  (contact) => contact.id !== deletedContactId,
+                ),
+              },
+            };
+            cache.writeQuery({ ...query, data });
+
+            push({
+              pathname: '/accountLists/[accountListId]/contacts',
+              query: { accountListId, searchTerm },
```

Comment:
Do we want to pass all except `contactDetailsId`? Might be more future proof.

Reply:
So something like ``` const { contactId: _, searchTerm, ...queryWithoutContactId } = query;``` and then 
```
push({
     pathname: '/accountLists/[accountListId]/contacts',
     query: { searchTerm, ...queryWithoutContactId },
});
```

Reply:
Ya. `searchTerm` should already be in `query` though.

Reply:
I extract it out for the query to do the cache update
```
const query = {
    query: ContactsDocument,
    variables: {
        accountListId,
        searchTerm,
    },
};
```